src/cryptsetup/cryptsetup-generator.c

```diff
@@ -387,10 +387,12 @@ static int create_disk(
                         );
         }
 
+        const char *target = netdev ? "remote-cryptsetup.target" : "cryptsetup.target";
+        if (in_initrd())
+                target = "initrd-cryptsetup.target";
```

Comment:
maybe merge these?

```c
const char *target = 
         in_initrd() ? "initrd-cryptsetup.target" :
              netdev ? "remote-cryptsetup.target" :
                       "cryptsetup.target"; 
```

Reply:
I actually initially did exactly that, though felt funny that we could get more legible code in the same or less number of lines.

Though actually, I like how the fstab-generator does this so I copied that instead. Using the same flow also helps to draw the parallel between the two.